### PR TITLE
Migrate APIM720GetAllEndPointsTestCase

### PR DESCRIPTION
**Pull request by vithu30**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java [hunk 1]
```diff
@@ -88,80 +88,62 @@ public APIM720GetAllEndPointsTestCase(TestUserMode userMode) {
...
         super.init(userMode);
         log.info("Test Starting user mode:" + userMode);
 
-        String storeURLHttp = storeUrls.getWebAppURLHttp();
-        apiStore = new APIStoreRestClient(storeURLHttp);
-
-        String publisherURLHttp = publisherUrls.getWebAppURLHttp();
-        apiPublisher = new APIPublisherRestClient(publisherURLHttp);
+        //publisher login
+        apiPublisher = new RestAPIPublisherImpl(publisherContext.getContextTenant().getContextUser().getUserName(),
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java]: We should be able to remove these lines as these are initialized in the base class.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java [hunk 2]
```diff
@@ -172,99 +154,43 @@ public void setEnvironment() throws Exception {
...
     public void destroy() throws Exception {
-        apiProvider = publisherContext.getContextTenant().getContextUser().getUserName();
-        HttpResponse deleteApiResponse = apiPublisher.deleteAPI(apiName, apiVersion, apiProvider);
-        assertEquals(deleteApiResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                     "Error in Delete API Response");
-        JSONObject deleteApiJsonObject = new JSONObject(deleteApiResponse.getData());
-        assertFalse(deleteApiJsonObject.getBoolean("error"), "Error in Delete API");
-        super.cleanUp();
+        apiPublisher.deleteAPI(apiID);
+        super.cleanUpUsingRest();
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java]: Lets use the previous cleanUp method as discussed.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java [hunk 3]
```diff
@@ -88,80 +88,62 @@ public APIM720GetAllEndPointsTestCase(TestUserMode userMode) {
...
-                     "Publisher Login Response Code is Mismatched: ");
-
         //store login
-        HttpResponse loginResponse = apiStore.login(storeContext.getContextTenant().getContextUser().getUserName(),
-                                                    storeContext.getContextTenant().getContextUser().getPassword());
-        assertEquals(loginResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                     "Response code is Mismatched in Login Response");
-        JSONObject loginJsonObject = new JSONObject(loginResponse.getData());
-        assertFalse(loginJsonObject.getBoolean("error"), "Response data error in Login Request");
+        apiStore = new RestAPIStoreImpl(storeContext.getContextTenant().getContextUser().getUserName(),
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java]: We can remove this.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java [hunk 4]
```diff
@@ -88,80 +88,62 @@ public APIM720GetAllEndPointsTestCase(TestUserMode userMode) {
...
-        HttpResponse statusUpdateResponse = apiPublisher.changeAPILifeCycleStatus(updateRequest);
-        assertEquals(statusUpdateResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                     "Response Code is Mismatched");
-
-        waitForAPIDeploymentSync(apiProvider, apiName, apiVersion, APIMIntegrationConstants.IS_API_EXISTS);
-
-        JSONObject statusUpdateJsonObject = new JSONObject(statusUpdateResponse.getData());
-        assertFalse(statusUpdateJsonObject.getBoolean("error"), "API is not published");
-
+        Thread.sleep(15000);
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java]: Why have we added this sleep? This is not the best practice.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java (hunk 1 above)
Comment (vithu30) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java]: Removed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java (hunk 2 above)
Comment (vithu30) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java]: Fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java (hunk 3 above)
Comment (vithu30) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java]: Removed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java (hunk 4 above)
Comment (vithu30) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java]: Fixed

**Pull request merged**